src/backend/distributed/commands/function.c

```diff
@@ -185,14 +197,29 @@ create_distributed_function(PG_FUNCTION_ARGS)
 			ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
 							errmsg("cannot distribute the function \"%s\" since the "
 								   "distribution argument is not valid ", functionName),
-							errhint("To provide \"colocate_with\" option, the"
-									" distribution argument parameter should also "
-									"be provided")));
+							errhint("To provide \"colocate_with\" option with a"
+									" distributed table, the distribution argument"
+									" parameter should also be provided")));
 		}
 
 		/* set distribution argument and colocationId to NULL */
 		UpdateFunctionDistributionInfo(&functionAddress, NULL, NULL);
 	}
+	else if (distributionArgumentName == NULL && colocationWithReferenceTable)
```

Comment:
this function seems to be already long, could be nice to create smaller functions for some blocks